packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-options.page.tsx

```diff
@@ -56,31 +56,32 @@ export const getServerSideProps: GetServerSideProps = async ({
     (section) => section.sectionId === sectionId
   )?.sectionTitle;
 
-  const result = await callServiceMethod(
-    req,
-    res,
-    async (body: any) => {
-      options = Object.keys(body).reduce((array, key) => {
-        if (key.startsWith('options')) {
-          array.push(...body[key]);
-        }
+  function processBody(body: any, options: string[]) {
+    options = Object.keys(body).reduce((array, key) => {
+      if (key.startsWith('options')) {
+        array.push(...body[key]);
+      }
 
-        if (key.startsWith('delete_')) {
-          const deleteOptionIndex = Number(key.split('_')[1]);
-          array.splice(deleteOptionIndex, 1);
-        }
-        return array;
-      }, [] as string[]);
+      if (key.startsWith('delete_')) {
+        const deleteOptionIndex = Number(key.split('_')[1]);
+        array.splice(deleteOptionIndex, 1);
+      }
+      return array;
+    }, [] as string[]);
+    return options;
```

Comment:
```suggestion
  return Object.keys(body).reduce((array, key) => {
      if (key.startsWith('options')) {
        array.push(...body[key]);
      } else if (key.startsWith('delete_')) {
        const deleteOptionIndex = Number(key.split('_')[1]);
        array.splice(deleteOptionIndex, 1);
      }
      return array;
    }, [] as string[]);
```
best practice to not directly modify parameters passed to a function unless there's a specific reason, plus using an else-if here reduces complexity and processing slightly